ntp-daemon/src/system.rs

```diff
@@ -0,0 +1,199 @@
+        let system_poll = global_system_snapshot.read().await.poll_interval;
+        let result = FilterAndCombine::run(config, &snapshots, ntp_instant, system_poll);
+
+        let clock_select = match result {
+            Some(clock_select) => clock_select,
+            None => {
+                info!("filter and combine did not produce a result");
+                continue;
+            }
+        };
+
+        let offset_ms = clock_select.system_offset.to_seconds() * 1000.0;
+        let jitter_ms = clock_select.system_jitter.to_seconds() * 1000.0;
+        info!(offset_ms, jitter_ms, "system offset and jitter");
+
+        let adjust_type = controller.update(
+            clock_select.system_offset,
+            clock_select.system_jitter,
+            clock_select.system_peer_snapshot.root_delay,
+            clock_select.system_peer_snapshot.root_dispersion,
+            clock_select.system_peer_snapshot.leap_indicator,
+            clock_select.system_peer_snapshot.time,
+        );
+
+        // Handle situations needing extra processing
+        match adjust_type {
+            ClockUpdateResult::Panic => {
+                panic!(
+                    r"Unusually large clock step suggested,
+                            please manually verify system clock and reference clock 
+                                 state and restart if appropriate."
+                )
+            }
+            ClockUpdateResult::Step => {
+                peers.reset_all();
+
+                reset_epoch = reset_epoch.inc();
+                reset_tx.send_replace(reset_epoch);
+            }
+            _ => {}
+        }
+
+        // Handle updating system snapshot
+        if let ClockUpdateResult::Ignore = adjust_type {
+            // ignore this update
+        } else {
+            let mut global = global_system_snapshot.write().await;
+            global.poll_interval = controller.preferred_poll_interval();
+            global.leap_indicator = clock_select.system_peer_snapshot.leap_indicator;
+        }
+    }
+
+    // the channel closed and has no more messages in it
+    Ok(())
+}
+
+#[derive(Debug, Clone, Copy)]
+enum PeerStatus {
+    /// This peer is demobilized, meaning we will not send further packets to it.
+    /// Demobilized peers are kept because our logic is built around using indices,
+    /// and removing a peer would mess up the indexing.
+    Demobilized,
+    /// We are waiting for the first snapshot from this peer _in the current reset epoch_.
+    /// This state is the initial state for all peers (when the system is spawned), and also
+    /// entered when the system performs a clock jump and forces all peers to reset, or when a peer
+    /// indicates that it is no longer fit for synchronization (e.g. root distance became too big)
+    ///
+    /// A peer can leave this state by either becoming demobilized, or by sending a snapshot that
+    /// is within the system's current reset epoch.
```

Comment:
which part?

Reply:
The current text still suggests this only happens in response to resets, even though there are other reasons this can happen (which are probably just as common)

Reply:
this state i mean

Reply:
o never mind, i cannot read